MetaMask/metamask-extension
# feat: add simulation preview for native balance changes

## **Description**
* Display simulation preview for native balance changes
* Storybook entries for native balance changes
* e2e test for a native balance change
* Code fence for `simulation-preview`

Related: "2nd PR" - https://github.com/MetaMask/metamask-extension/pull/23529

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23503?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2141

## **Manual testing steps**

1. Start sending some ETH to any address.
2. On the confirmation page you will see a simulation preview.

## **Screenshots/Recordings**



### **Before**



### **After**

<img width="519" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/507015/4820d0ba-ccb3-4063-8ce1-f262b623cfae">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
